### PR TITLE
feat(fastify): update to v4

### PR DESCRIPTION
**Pull request by dan-online**
## Changes

- [x] Change CI to node@16
- [x] Remove ``noImplicitAny``
- [x] Update for fastify@4.0.1
- [x] Bump ``ava`` ``fastify`` ``typescript`` ``eslint``

Feel free to review and request any changes needed :)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (yonathan06) [inline .vscode/settings.json:1]: This shouldn't be in the repo, can you please add it to `.gitignore`?

File: index.ts [hunk 2]
```diff
@@ -12,22 +12,14 @@ import type {
...
-  POST = 'post',
-  PUT = 'put',
-  DELETE = 'delete',
-  OPTIONS = 'options',
-  PATCH = 'patch',
-  HEAD = 'head',
-  CONNECT = 'connect',
-  TRACE = 'trace',
-}
+const methods = ['DELETE', 'GET', 'HEAD', 'PATCH', 'POST', 'PUT', 'OPTIONS'];
```
Comment (yonathan06) [inline index.ts:19]: I see that HTTPMethods is a type and not an enum (unfortunately)
Can you add the type for this array? e.g. `const methods: HTTPMethods[]`?

**Review by yonathan06 — COMMENTED** [review]
Thank you for the PR 🙏
I added some comments
I also see the tests are not triggered, can you please add `pull_request ` to the `on` field?

**Review by yonathan06 — APPROVED** [review]

**Comment by yonathan06** [comment]
Thank you @dan-online for the change!
I will create a new release with a new major version 3.0.0 to keep version 2.x.x compatible with fastify 3

**Comment by yonathan06** [comment]
@dan-online now published under v3.0.0

**Comment by dan-online** [comment]
Awesome thanks for the quick response!

**Pull request merged**